certbot: 0.39.0 -> 1.0.0

flokli opened the pull request:
This bumps certbot from 0.39.0 to their latest release, 1.0.0.

I also got the unit tests to work again :tada:. We don't need to pass in pebble, or run the certbot-ci tests, as elaborated in https://github.com/certbot/certbot/issues/7450.

~~This is still pending on https://github.com/zenhack/simp_le/issues/131, as `simp_le` currently doesn't want to build with `acme-1.0.0`.~~

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

flokli committed:
certbot: 0.39.0 -> 1.0.0

FRidh commented on pkgs/tools/admin/certbot/default.nix:
```diff
@@ -35,6 +34,7 @@ python37Packages.buildPythonApplication rec {
   ];
 
   buildInputs = [ dialog ] ++ (with python37Packages; [ mock gnureadline ]);
+  nativeBuildInputs = [ python37Packages.setuptoolsCheckHook ];
```
as long as `format = "setuptools";` is passed to `buildPython*` (its the default), this hook is included.

flokli committed:
simpl_le: 0.16.0 -> 0.17.0

flokli commented:
@GrahamcOfBorg test acme

flokli commented:
`simpl_le` was bumped to a version compatible with our `acme` python module, and I got the acme vm test to past, marked as "Ready for review".

Br1ght0ne reviewed (commented):
Checked with `nixpkgs-review`, found out that `hass-nabucasa` can't build because of it depends on `acme==0.39.0` (strongly).

```
[7 built (2 failed), 155 copied (201.4 MiB), 29.9 MiB DL]
error: build of '/nix/store/18cjv2hq4791vrvnwvhv06m0f1xplw2q-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/75663
2 package failed to build:
python37Packages.hass-nabucasa python38Packages.hass-nabucasa

5 package built:
certbot python27Packages.acme python37Packages.acme python38Packages.acme simp_le
```

flokli commented:
Well, there was https://github.com/NabuCasa/hass-nabucasa/pull/116/files, so I assume porting to a later version probably won't be too complicated either ;-
@filalex77 could you open a PR upstream?

NinjaTrappeur commented:
Ping @Scriptkiddi, maintainer of Hass. Could you have a look? We really need to bump Certbot.

makefu commented:
@NinjaTrappeur @flokli we could simply make the hass-nabucasa update part of this PR, also the question is if the pin to 0.39 is really necessary (it is not according to the PR linked).
So there are two choices:
1. update hass-nabucasa to the latest dev release
2. remove the acme version pin with sed in the patchPhase

i would prefer 2

makefu commented on pkgs/tools/admin/certbot/default.nix:
```diff
@@ -55,15 +53,7 @@ python37Packages.buildPythonApplication rec {
-  '';
-
-  dontUseSetuptoolsCheck = true;
+  doCheck = true;
```
i love it when tests get re-enabled for packages :+1:

flokli replied:
Yes, I'm also feeling much more confident with tests being enabled.

makefu reviewed (approved):
haven't tested it but code looks good to me!

flokli committed:
hass-nabucasa: relax version bounds

upstreamed in https://github.com/NabuCasa/hass-nabucasa/pull/119, but
doesn't appply cleanly

flokli commented:
@makefu relaxed the version bounds for hass-nabucasa, tests still pass. Also upstreamed to https://github.com/NabuCasa/hass-nabucasa/pull/119 .

Scriptkiddi commented:
lgtm

makefu commented:
@ofborg build python38Packages.hass-nabucasa

makefu commented:
@ofborg build python37Packages.hass-nabucasa
tested the issue, seems like yarl does not build with python38 (but python37)

makefu commented:
For the bump of yarl i used the following code in pkgs/development/python-modules/yarl/default.nix :
```nix
{ stdenv
, fetchPypi
, buildPythonPackage
, multidict
, pytest
, pytestcov
, idna
}:

buildPythonPackage rec {
  pname = "yarl";
  version = "1.4.2";

  src = fetchPypi {
    inherit pname version;
    sha256 = "0jzpgrdl6415zzl8js7095q8ks14555lhgxah76mimffkr39rkaq";
  };
  checkPhase = ''
    pytest ./tests ./yarl
  '';

  checkInputs = [ pytest pytestcov ];
  propagatedBuildInputs = [ multidict idna ];

  meta = with stdenv.lib; {
    description = "Yet another URL library";
    homepage = https://github.com/aio-libs/yarl/;
    license = licenses.asl20;
    maintainers = with maintainers; [ dotlambda ];
  };
}
```

flokli commented:
Some of yarl's test being broken isn't really connected to this PR.

Let's track this upstream at https://github.com/aio-libs/yarl/issues/410 and merge this PR, it;s already broken before.
